sound/soc/sdca/sdca_asoc.c

```diff
@@ -863,43 +855,160 @@ static int control_limit_kctl(struct device *dev,
 	max = sign_extend32(max, control->nbits - 1);
 
 	/*
-	 * FIXME: Only support power of 2 step sizes as this can be supported
-	 * by a simple shift.
+	 * The SDCA volumes are in steps of 1/256th of a dB, so we need to
+	 * scale them to 1/100ths of a dB for the ALSA TLV.
+	 * The min and max values are in Q7.8 format, so we need to scale
+	 * them to 1/100ths of a dB.
 	 */
-	if (hweight32(step) != 1) {
-		dev_err(dev, "%s: %s: currently unsupported step size\n",
-			entity->label, control->label);
-		return -EINVAL;
-	}
-
-	/*
-	 * The SDCA volumes are in steps of 1/256th of a dB, a step down of
-	 * 64 (shift of 6) gives 1/4dB. 1/4dB is the smallest unit that is also
-	 * representable in the ALSA TLVs which are in 1/100ths of a dB.
-	 */
-	shift = max(ffs(step) - 1, 6);
+	min = min * 100 / SCALE_FACTOR;
+	max = max * 100 / SCALE_FACTOR;
+	step = step * 100 / SCALE_FACTOR;
```

Comment:
I would be tempted to bracket the multiplies even though it isn't strictly necessary but it does make it more clear.

Reply:
Sure, will add a definition for it.